source/lib/compiler.ts

```diff
@@ -28,21 +28,26 @@ const expectErrordiagnosticCodesToIgnore = new Set<DiagnosticCode>([
 	DiagnosticCode.PropertyMissingInType1ButRequiredInType2
 ]);
 
+type IgnoreDiagnosticResult = 'preserve' | 'ignore' | Location;
+
 /**
  * Check if the provided diagnostic should be ignored.
  *
  * @param diagnostic - The diagnostic to validate.
  * @param expectedErrors - Map of the expected errors.
  * @returns Boolean indicating if the diagnostic should be ignored or not.
  */
-const ignoreDiagnostic = (diagnostic: TSDiagnostic, expectedErrors: Map<Location, any>): boolean => {
+const ignoreDiagnostic = (
+	diagnostic: TSDiagnostic,
+	expectedErrors: Map<Location, ExpectedError>
+): IgnoreDiagnosticResult => {
```

Comment:
Can you update the `@returns` in the TS doc?

Reply:
done